### DISCUSSION TITLE
How Can I Set a Custom Library Search Path, i.e `-L$HOME/.local/lib`?

### DISCUSSION THREAD

**Discussion by zicklag** [ocaml/opam#5366] [Q&A] [answered]
Hey there!

I have a manually installed version of gmp, which is needed to install coq, but zarith fails to install because it can't find it. ( I'm not root on my machine so I can't just install the `libgmp-dev` package. )

I started by exporting a `CFLAGS` environment variable, which worked for the first package, before getting stuck on zarith because it used `-lgmp`, but didn't add my `-L` flag to set an extra library search dir.

Is there any way to set default extra flags for `ocamlmklib` through environment variables or a config file?

**Comment by zicklag** [comment, marked as answer]
Ah, nevermind, I found it. I had to set `CPPFLAGS` and `LDFLAGS`, not just `CFLAGS`.

**Comment by dbuenzli** [comment]
Try e.g.: 
```
export LIBRARY_PATH=/usr/local/lib:$LIBRARY_PATH
export C_INCLUDE_PATH=/usr/local/include:$C_INCLUDE_PATH
```

See https://github.com/ocaml/opam/issues/3049

**Reply by zicklag** [reply]
Oh, thanks for the quick reply! I just saw you answer as I found that setting the `LDFLAGS` fixed it at least in this case. Knowing about the other variables may help if the ones I set now don't work for other packages.

Thanks again!
